src/jemalloc/include/jemalloc/internal/jemalloc_internal.h.in

```diff
@@ -1041,7 +1041,7 @@ iralloct(void *ptr, size_t size, size_t extra, size_t alignment, bool zero,
 			}
 		}
 
-		return NULL;
+		return (ret != NULL) ? (ret) : (ptr);
```

Comment:
In the same vein, you could argue that realloc never fails while shrinking. It just doesn't do anything sometimes.

I don't know, I guess it should return a the same ptr if we did in fact resize something (but maybe not to the desired allocation class)?